### ISSUE TITLE
Catmull-rom interpolation smoothness

### ISSUE THREAD

**Issue by Cheryl-Lao** [dilevin/computer-graphics-kinematics#24]
Should our animation look as smooth as the one in the README's gif? Mine gets to all the poses but it looks a bit more jumpy (as if there are frames missing compared to the README version)

Also, #20 says that we don't need to loop the animation, so is it okay if the whole arm just disappears for a second before repeating?

**Comment by gabriellemadden** [comment]
mine is also jumpy, i *hope* its just my libgl lagging out because it definitely doesnt look as smooth as the animation!!

I had the issue where my whole arm disappeared at the end of the animation too, but I was able to actually fix that part so make sure you are handling the end edge case of the spline properly as in #20

**Comment by Cheryl-Lao** [comment]
Hmmm I tried the tips in #20 but some of the bones still disappear when it loops :'( When it gets to the upright rotation part, all of the bones (except the bottom two) disappear

**Comment by Cheryl-Lao** [comment]
@gabriellemadden I actually ended up trying to build on release mode and it fixed the jumpiness! No clue how that made a difference, but it worked!

**Comment by gabriellemadden** [comment]
> Hmmm I tried the tips in #20 but some of the bones still disappear when it loops :'( When it gets to the upright rotation part, all of the bones (except the bottom two) disappear

Yeah thats the same thing that happened to me. I followed the catmull-rom tangents from the lecture slides and I ended up doing a separate case for when t is between the second last and last keyframe, looping back to the first frame. If you're using the tangent equation from wikipedia I don't think what I did would apply though..
